### ISSUE TITLE
Bug: PRNG key reuse in autodiff lecture + unused import in numpy_vs_numba_vs_jax

### ISSUE THREAD

**Issue by mmcky** [QuantEcon/lecture-python-programming.myst#529]
## Summary

Two minor issues found during translation review of the zh-cn repo (via Copilot code review on [QuantEcon/lecture-python-programming.zh-cn#19](https://github.com/QuantEcon/lecture-python-programming.zh-cn/pull/19)):

### 1. PRNG key reuse in `lectures/autodiff.md`

In the gradient descent / simulated data section, the same `key` is used for both `jax.random.uniform` and `jax.random.split`:

```python
x = jax.random.uniform(key, (n,))

α, β, σ = 0.5, 1.0, 0.1
key, subkey = jax.random.split(key)
ϵ = jax.random.normal(subkey, (n,))
```

This violates JAX's guidance that keys should be treated as single-use. `key` is consumed by `uniform` and then reused as input to `split`, which can lead to correlated draws.

**Suggested fix:**
```python
key, x_key, eps_key = jax.random.split(key, 3)
x = jax.random.uniform(x_key, (n,))

α, β, σ = 0.5, 1.0, 0.1
ϵ = jax.random.normal(eps_key, (n,))
```

### 2. Unused `random` import in `lectures/numpy_vs_numba_vs_jax.md`

```python
import random
```

`random` is imported but never used in the lecture. Can be safely removed.

---

Found by GitHub Copilot code review during automated translation sync.
